src/main/java/com/uid2/shared/secure/GcpOidcAttestationProvider.java

```diff
@@ -0,0 +1,95 @@
+package com.uid2.shared.secure;
+
+import com.uid2.shared.secure.gcpoidc.*;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+public class GcpOidcAttestationProvider implements IAttestationProvider{
+    private static final Logger LOGGER = LoggerFactory.getLogger(GcpOidcAttestationProvider.class);
+
+    private final ITokenSignatureValidator tokenSignatureValidator;
+
+    private final List<IPolicyValidator> supportedPolicyValidators;
+
+    private final Set<String> allowedEnclaveIds = new HashSet<>();
+
+    public GcpOidcAttestationProvider(){
+        this(new TokenSignatureValidator(), Arrays.asList(new PolicyValidator()));
+    }
+
+    // used in UT
+    protected GcpOidcAttestationProvider(ITokenSignatureValidator tokenSignatureValidator, List<IPolicyValidator> supportedPolicyValidators){
+        this.tokenSignatureValidator = tokenSignatureValidator;
+        this.supportedPolicyValidators = supportedPolicyValidators;
+    }
+
+    @Override
+    public void attest(byte[] attestationRequest, byte[] publicKey, Handler<AsyncResult<AttestationResult>> handler) {
+        try {
+            var tokenString = new String(attestationRequest, StandardCharsets.US_ASCII);
+
+            LOGGER.debug("Validating signature...");
+            var tokenPayload = tokenSignatureValidator.validate(tokenString);
+
+            if(this.Validate(tokenPayload)){
+                handler.handle(Future.succeededFuture(new AttestationResult(publicKey)));
```

Comment:
validation isn't async, is there any need to handle it asynchronously?

Reply:
1. Token signature validation:  involves API call to fetch GCP public key, but it will maintain a local cache. 
2. Token payload validation:  just some in-proc calculation. Should be super quick.

Let's keep it as is.